### DISCUSSION TITLE
Avalonia how to get the BitmapId of AvaloniaResource

### DISCUSSION THREAD

**Discussion by zimuyanghua** [Mapsui/Mapsui#2606] [Q&A]
When the image is set to AvaloniaResource, how do I get the BitmapId?
![image](https://github.com/Mapsui/Mapsui/assets/46125819/3423f319-9d6c-47c3-b113-860b40d7b63c)
![image](https://github.com/Mapsui/Mapsui/assets/46125819/f950d3bf-b25c-4d8c-b0f1-cc2483407818)

**Comment by zimuyanghua** [comment]
Can it only be set to EmbeddedResource?
![image](https://github.com/Mapsui/Mapsui/assets/46125819/4aad8a95-ac4d-4a53-9be9-3e6f933ee1b3)

**Reply by inforithmics** [reply]
Yes currently  only embedded Resource works.

**Reply by inforithmics** [reply]
The Resource (MemoryStream/Image) has to be registered in BitmapRegistry.Instance.Register() which returns the BitmapId.

**Reply by zimuyanghua** [reply]
OK, Thanks

**Comment by zimuyanghua** [comment]
another problem, using Chinese text does not display
![image](https://github.com/Mapsui/Mapsui/assets/46125819/2cee4011-7fb8-456a-a6d5-0509f59e1ea1)

**Reply by inforithmics** [reply]
<img width="90" alt="image" src="https://github.com/Mapsui/Mapsui/assets/2290863/682617b2-1fd2-4e36-85a0-e086943ab30d">

I tried in Avalonia and chinese characters seem to work. Can you provide a sample where it doesn't work?

**Reply by zimuyanghua** [reply]
Ok, I uploaded it here, please take a look.
https://github.com/zimuyanghua/MapsuiDemo
